doc/release-notes.md

```diff
@@ -318,7 +97,7 @@ Dash Core tree 0.12.1.x was a fork of Bitcoin Core tree 0.12.
 
 These release are considered obsolete. Old release notes can be found here:
 
-- [v0.17.0.3](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.17.0.3.md) released June/07/2021
```

Comment:
should this v0.17.0.3 line be not removed to be history ?

Reply:
@thelazier I believe here we only keep the last version's release notes, right @PastaPastaPasta ?

Reply:
All previous releases should be listed here